resources/stsci.edu/schemas/spline1d-1.0.0.yaml

```diff
@@ -0,0 +1,61 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/asdf/transform/spline1d-1.0.0"
+tag: "tag:stsci.edu:asdf/transform/spline1d-1.0.0"
+title: >
+  A spline1d model.
+
+description: |
+  A spline1d model represented by an array of its knots, an array of its
+  coefficients, and its degree. In addition the bounding endpoints
+  of the spline can be represented as well
+
+examples:
+  -
+    - Non-bounded spline
+    - |
+        !transform/spline1d-1.0.0
+          knots: !core/ndarray-1.0.0
+                          [0.0, 0.5, 1]
+          coefficients: !core/ndarray-1.0.0
+                          [1.2, 0.3, 56.1]
+          degree: 1
+  -
+    - Bounded spline
+    - |
+        !transform/spline1d-1.0.0
+          knots: !core/ndarray-1.0.0
+                          [0.0, 0.5, 1]
+          coefficients: !core/ndarray-1.0.0
+                          [1.2, 0.3, 56.1]
+          degree: 1
+          spline_bounds: [-0.5, 1.5]
+
+definitions:
+  coeffs:
+    description: |
+      An array of values
+    anyOf:
+      - tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
+      - tag: tag:stsci.edu:asdf/unit/quantity-1.1.0
+      - type: array
```

Comment:
IMO it's preferable to have a single option for serialization so that life will be simpler for anybody who has to implement a reader for these files later (particularly for users of statically typed languages who will find this kind of thing annoying).  Is there a way we can achieve that here?  Some ideas:

- Since ndarray allows inlined arrays anyway, maybe we don't need a simple array?
- On the other hand, will there ever be enough coefficients to be worth storing in a binary block?  Maybe we drop ndarray and only allow array.
- Is it possible to construct a quantity with unit = none or something like that?